toolkit/tools/specreader/specreader.go

```diff
@@ -410,6 +423,7 @@ func parseProvides(rpmsDir, srpmPath string, list []string) (providerlist []*pkg
 
 		if listEntry[tag] == "rpm" {
 			logger.Log.Trace("rpm ", listEntry[value])
+			// We will decide if this is a toolchain package later, and update the path if so.
```

Comment:
Man, this is a terrible function to read. Am I getting this right, that on one loop we'll set `rpmPath` and then we expect another run where the tag will be "provides", so we can use the `rpmPath` we just set there? Are we even guaranteed to end with a "provides" for each `rpmPath` we set?

Reply:
Will give this some thought, might return later to clean it up.

Reply:
I think I'll leave this as is for now, this function needs a re-write for sure but I'd rather not do that in this PR.